File: 70.climbing-stairs.py
#
# @lc app=leetcode id=70 lang=python3
#
# [70] Climbing Stairs
#

# @lc code=start
class Solution:
    from functools import lru_cache
    @lru_cache(None)
    def f(self, x, y):
        if x == y:
            return 1
        elif x > y:
            return 0;
        return self.f(x + 1, y) + self.f(x + 2, y)
        
    def climbStairs(self, n: int) -> int:
        return self.f(0, n)
# @lc code=end

File: 70.climbing-stairs.cpp
/*
 * @lc app=leetcode id=70 lang=cpp
 *
 * [70] Climbing Stairs
 */
#include "iostream"

// @lc code=start
class Solution {
public:
    int climbStairs(int n) {
        long long a = 1, b = 1;
        while (n--) {
            a = (b += a) - a;
        }
        return a;
    }
};
// @lc code=end

